fjord/suggest/providers/trigger/tests/test_models.py

```diff
@@ -0,0 +1,189 @@
+from fjord.base.tests import TestCase
+from fjord.feedback.tests import ProductFactory, ResponseFactory
+from fjord.suggest.providers.trigger.models import (
+    TriggerRule,
+    _generate_keywords_regex
+)
+from fjord.suggest.providers.trigger.tests import TriggerRuleFactory
+
+
+class TriggerRuleCleanTests(TestCase):
+    def test_clean_locales(self):
+        tests = [
+            # testcase, expected
+            ('', ''),
+            ('en-US', 'en-US'),
+            ('  en-US  ', 'en-US'),
+            ('  es, fr,  ', 'es,fr'),
+        ]
+
+        for testcase, expected in tests:
+            assert TriggerRule.clean_locales(testcase) == expected
+
+    def test_clean_versions(self):
+        tests = [
+            # testcase, expected
+            ('', ''),
+            ('38', '38'),
+            ('37*,38*', '37*,38*'),
+            (' 37*, 38*, ', '37*,38*'),
+        ]
+
+        for testcase, expected in tests:
+            assert TriggerRule.clean_versions(testcase) == expected
+
+    def test_clean_keywords(self):
+        tests = [
+            # testcase, expected
+            ('', ''),
+            ('a', 'a'),
+            ('a\nb\nc', 'a\nb\nc'),
+            ('a\n\nb', 'a\nb'),
+            ('a\n  \nb', 'a\nb'),
+            (' a \n b \n\n c ', 'a\nb\nc'),
+        ]
+        for testcase, expected in tests:
+            assert TriggerRule.clean_keywords(testcase) == expected
+
+
+class TriggerRuleMatchTests(TestCase):
+    def test_match_locale(self):
+        tests = [
+            # tr locales, feedback locale, expected
+            ('', '', True),
+            ('', 'en-US', True),
+            ('', 'fr', True),
+
+            ('en-US', '', False),
+            ('en-US', 'en-US', True),
+            ('en-US', 'fr', False),
+
+            ('en-US,fr', '', False),
+            ('en-US,fr', 'en-US', True),
+            ('en-US,fr', 'fr', True),
+            ('en-US,fr', 'es', False)
+        ]
+        for tr_locales, feedback_locale, expected in tests:
+            tr = TriggerRuleFactory(locales=tr_locales)
+            assert tr.match_locale(feedback_locale) == expected
+
+    def test_match_versions(self):
+        tests = [
+            # tr versions, feedback version, expected
+            ('', '', True),
+            ('', '38', True),
+            ('', '38.0.5', True),
+
+            ('38', '', False),
+            ('38', '38', True),
+            ('38', '38.0.5', False),
+
+            ('38*', '', False),
+            ('38*', '38', True),
+            ('38*', '38.0.5', True),
+
+            ('38*,37.1', '', False),
+            ('38*,37.1', '38', True),
+            ('38*,37.1', '38.0.5', True),
+            ('38*,37.1', '37', False)
```

Comment:
I think you should add a case here `('38*,37.1', '37.1', True)`, to make sure it can match with not-the-first version.